common-pg/src/main/scala/com/socrata/pg/analyzer2/Sqlizer.scala

```diff
@@ -0,0 +1,585 @@
+
+        (recompressedLeftDoc, recompressedRightDoc, recompressedLeftSchema)
+      }
+
+    (effectiveLeftDoc.parenthesized +#+ sqlizeOp(op) +#+ effectiveRightDoc.parenthesized, effectiveSchema)
+  }
+
+  private def recompress(stmt: Doc, schema: AugmentedSchema, mask: Set[ColumnLabel]): (Doc, AugmentedSchema) = {
+    val tmpTable = Doc(namespace.gensym())
+    val schemaOrder = schema.keys.toSeq
+
+    val selectionDoc = Seq(
+      Some(d"SELECT"),
+      Some(
+        schemaOrder.flatMap { columnLabel =>
+          val typ = schema(columnLabel).typ
+          val rep = repFor(typ)
+
+          if(mask(columnLabel)) {
+            val underlyingColumnSqls = rep.expandedDatabaseColumns(columnLabel).map { dbColName =>
+              tmpTable ++ d"." ++ Doc(dbColName)
+            }
+            Seq(ExprSql.compressSqls(underlyingColumnSqls) +#+ d"AS" +#+ Doc(rep.compressedDatabaseColumn(columnLabel)))
+          } else if(schema(columnLabel).isExpanded) {
+            rep.expandedDatabaseColumns(columnLabel).map { dbColName =>
+              tmpTable ++ d"." ++ Doc(dbColName) +#+ d"AS" +#+ Doc(dbColName)
+            }
+          } else {
+            Seq(tmpTable ++ d"." ++ Doc(rep.compressedDatabaseColumn(columnLabel)) +#+ d"AS" +#+ Doc(rep.compressedDatabaseColumn(columnLabel)))
+          }
+        }.commaSep
+      ).filter { _ => schemaOrder.nonEmpty }
+    ).flatten.vsep.nest(2)
+    val fromDoc = (d"FROM" ++ Doc.lineSep ++ stmt.parenthesized +#+ d"AS" +#+ tmpTable).nest(2)
+
+    val doc = Seq(
+      selectionDoc,
+      fromDoc
+    ).vsep.group
+
+    val newSchema = OrderedMap() ++ schema.map { case (lbl, augTyp) =>
+      if(mask(lbl)) lbl -> augTyp.copy(isExpanded = false)
+      else lbl -> augTyp
+    }
+
+    (doc, newSchema)
+  }
+
+  private def sqlizeOp(op: TableFunc): Doc =
+    op match {
+      case TableFunc.Union => d"UNION"
+      case TableFunc.UnionAll => d"UNION ALL"
+      case TableFunc.Intersect => d"INTERSECT"
+      case TableFunc.IntersectAll => d"INTERSECT ALL"
+      case TableFunc.Minus => d"MINUS"
+      case TableFunc.MinusAll => d"MINUS ALL"
+    }
+
+  private def sqlizeValues(
+    values: Values,
+    availableSchemas: AvailableSchemas,
+    topLevel: Boolean
+  ): (Doc, AugmentedSchema) = {
+    val physicalSchema = values.schema.iterator.flatMap { case (label, nameEntry) =>
+      repFor(nameEntry.typ).expandedDatabaseColumns(label)
+    }
+
+    val exprSqlizer = new ExprSqlizer(this, availableSchemas, Vector.empty)
+
+    // Ugh, why are valueses so weird?
+    // Ok so, if we have a single row, just sqlize it as a FROMless SELECT.
+    if(values.values.tail.isEmpty) {
+      val row = values.values.head.iterator.map(exprSqlizer.sqlize).toVector
+
+      val physicalSchema = (values.labels, row).zipped.flatMap { (label, expr) =>
+        val rep = repFor(expr.typ)
+        expr match {
+          case _ :ExprSql.Compressed[MT] =>
```

Comment:
```suggestion
          case _ : ExprSql.Compressed[MT] =>
```
I realize this probably doesn't matter, but it bugs me.